tsimx6 add mwifiex external module

markfeathers opened the pull request.

markfeathers committed:
tasks: mwifiex-external-module: initial commit

Adds support for mwifiex / nxpwifi for iw61x

markfeathers committed:
tasks: imx-firmware: initial commit of NXP wifi firmware

markfeathers committed:
tasks: murata-nxp-linux-calibration: Added Murata NXP linux calibration firmware

markfeathers committed:
configs: tsimx9: Updated for Murata type 2dl wifi on 9370 P4

markfeathers committed:
configs: tsimx6: Added mwifiex driver support

markfeathers committed:
configs: tsimx6: Added mwifiex driver support

ts-kris reviewed (changes requested):
See previous comment.

ts-kris commented:
@markfeathers Is this PR still valid and waiting rework, or should it be closed?

markfeathers committed:
local_packages: Run depmod manually for the installed kernel

When we have more than one kernel module installed, they build their
modules simultaneously and each run depmod. When their output is installed,
there is a race to who's depmod database is installed.

To solve this, we add kmod to the vm image and run depmod manually after
package installation.

markfeathers commented:
This has been cleaned up, the previous comments should be addressed now.

ts-kris commented on configs/tsimx6_debian_13_x11_defconfig:
```diff
@@ -31,6 +37,5 @@ CONFIG_DS_USER_ROOT_EN=y
 CONFIG_DS_PURGE_LOCALES=y
 CONFIG_DS_HOSTNAME="tsimx6"
 CONFIG_DS_TIMEZONE="America/Phoenix"
-CONFIG_DS_DOWNGRADE_WPA_HOSTAPD=y
```
Other defconfigs don't remove this, and, removing this could be detrimental to plats still with silex (unless this issue was finally resolved?).

markfeathers replied:
The wpa downgrade task doesn't compile on debian 13 at the moment.  While that needs to be addressed, it is not this PR.  I could make a single separate config commit that fixes the incompatible DOWNGRADE configs so they dont keep showing up like this.

ts-kris commented on configs/tsimx6_debian_13_headless_defconfig:
```diff
@@ -26,6 +32,5 @@ CONFIG_DS_USER_ROOT_EN=y
 CONFIG_DS_PURGE_LOCALES=y
 CONFIG_DS_HOSTNAME="tsimx6"
 CONFIG_DS_TIMEZONE="America/Phoenix"
-CONFIG_DS_DOWNGRADE_WPA_HOSTAPD=y
```
Same as above.

ts-kris commented on tasks/core/local_packages/install-packages.sh:
```diff
@@ -45,4 +45,9 @@ install -d "$TARGET_PACKAGE_DIR"
 
+for module_dir in "$ROOTFS"/lib/modules/*; do
+    [[ -d "$module_dir" ]] || continue
+    depmod -b "$ROOTFS" "${module_dir##*/}"
```
Its kind of surprising that depmod isn't run as part of debian/ubuntu bootstrap

markfeathers replied:
Until we switch to the planned sprout + proper debian kernel packages, debian does not manage it for us.

markfeathers committed:
configs: normalize defconfigs with savedefconfig

Some configs were edited by hand it seems, and some Kconfig option
changes have affected some defaults.
